### PR TITLE
Add metric for no route to host errors

### PR DESCRIPTION
**Pull request by raiju**
## After this PR
==COMMIT_MSG==
Add metric for no route to host errors
==COMMIT_MSG==

### CODE REVIEW

File: dialogue-apache-hc5-client/src/main/metrics/dialogue-apache-client-metrics.yml [hunk 1]
```diff
@@ -40,6 +40,11 @@ namespaces:
         tags: [client-name, client-type]
         docs: Reports the time spent creating a new connection. This includes both connecting the socket and the full TLS handshake.
 
+      connection.route.error:
```
Comment (carterkozak) [inline dialogue-apache-hc5-client/src/main/metrics/dialogue-apache-client-metrics.yml]: Perhaps match the `connection.create` name with something like `connection.create.error`?

File: dialogue-apache-hc5-client/src/main/metrics/dialogue-apache-client-metrics.yml [hunk 2]
```diff
@@ -40,6 +40,11 @@ namespaces:
...
+        type: meter
+        tags: [client-name]
```
Comment (carterkozak) [inline dialogue-apache-hc5-client/src/main/metrics/dialogue-apache-client-metrics.yml]: let's tag with `cause` and use the throwable simple name

Comment (raiju) [reply dialogue-apache-hc5-client/src/main/metrics/dialogue-apache-client-metrics.yml]: I also added `client-type` to match the other connection metrics

File: dialogue-apache-hc5-client/src/test/java/com/palantir/dialogue/hc5/ApacheHttpClientChannelsTest.java [hunk 3]
```diff
@@ -143,16 +145,50 @@ public void countsUnknownHostExceptions() throws Exception {
             ListenableFuture<Response> future =
                     channel.execute(TestEndpoint.GET, Request.builder().build());
 
-            try (Response response = Futures.getChecked(future, UnknownHostException.class)) {
+            try (Response response = Futures.getUnchecked(future)) {
                 fail("This request should have failed with an unknown host exception! (code: %d)", response.code());
-            } catch (UnknownHostException _exception) {
-                // No-op, this is expected
+            } catch (UncheckedExecutionException exception) {
+                assertThat(exception.getCause()).isInstanceOf(UnknownHostException.class);
```
Comment (raiju) [inline dialogue-apache-hc5-client/src/test/java/com/palantir/dialogue/hc5/ApacheHttpClientChannelsTest.java:151]: Was a silly mistake of mine; `getChecked` wraps the exception (`converting most exceptions to a new instance of the given checked exception type`). This ensures it's of the expected type.

**Review by carterkozak — APPROVED** [review]

**Pull request merged**